internal/testtypes/builders.go

```diff
@@ -523,3 +544,20 @@ func setNestedRef(obj, refObj *unstructured.Unstructured, fields ...string) erro
 	}
 	return unstructured.SetNestedField(obj.UnstructuredContent(), ref, fields...)
 }
+
+// setSpecFields sets fields in an unstructured object from a map.
+func setSpecFields(obj *unstructured.Unstructured, fields map[string]interface{}) {
+	for k, v := range fields {
+		fieldParts := strings.Split(k, ".")
+		if len(fieldParts[0]) == 0 {
+			return
+		}
+		if fieldParts[0] != "spec" {
+			panic(fmt.Errorf("can not set fields outside spec"))
+		}
+		// TODO: Update to use new functions in the internal/contract package
```

Comment:
```suggestion
```
Not 100% sure, but I think we don't have (and don't need) a func in contract to set arbitrary fields